oc_cdtapi/DmsGetverAPI.py

```diff
@@ -46,6 +93,92 @@ def create_distr_request(self, version=None, source_version=None, distr_type=Non
 
         return distr_state_info
 
+    def download_file(self, image_id):
+        """
+        Downloads specified image to file
+        :param str image_id: id of image to download
+        :return: path to downloaded file
+        """
+        logging.debug('Reached download_file')
+        logging.debug('image_id: [%s]' % image_id)
+        logging.debug('Unsetting auth')
+        self.web.auth = None
+        headers = self.get_headers()
+        url = posixpath.join('api', 'v1', 'images', image_id, 'download')
+        logging.debug('url: [%s]' % url)
+        tf = tempfile.NamedTemporaryFile(delete=False)
+        tfn = tf.name
+        logging.debug('tf: [%s]' % tfn)
+        r = self.get(url, headers=headers, stream=True)
+        for chunk in r.iter_content(chunk_size=8192):
+            tf.write(chunk)
+        tf.seek(0)
+        return tfn
+
+    def get_audit(self, audit_id):
+        """
+        Requests audit record by record id
+        :param str audit_id: id of audit record
+        :return: audit record
+        """
+        logging.debug('Reached get_audit')
+        logging.debug('audit_id: [%s]' % audit_id)
+        self.web.auth = None
+        self.raise_exception_high = 499
+        headers = self.get_headers()
+        url = posixpath.join('api', 'v1', 'audit', audit_id)
+        r = self.get(url, headers=headers)
+        return self.json_or_none(r)
+
+    def get_headers(self):
+        """
+        """
+        logging.debug('Reached get_headers')
+        logging.debug('auth_token: [%s]' % self.auth_token)
+        headers = {'Accept': 'application/json; charset=utf-8', 'Authorization': f'Bearer {self.auth_token}'}
+        logging.debug('About to return: [%s]' % headers)
+        return headers
+
+    def search_image(self, version=None, distr_type=None):
+        """
+        searches for images, calls download
+        :return: path to temp file, image data
+        """
+        logging.debug('Reached download_image')
+        logging.debug('Unsetting auth')
+        self.web.auth = None
+        logging.debug('version = [%s]' % version)
+        logging.debug('distr_type = [%s]' % distr_type)
+        
+        url = posixpath.join('api', 'v1', 'images')
+        headers = self.get_headers()
+        params = {
+            'strict_filters': 'true',
+            'product_type': distr_type.lower(),
+            'product_version': version,
+            'product_release_stage': 'release',
+            'remap_tablespaces': 'true'
+        }
+        resp = self.get(url, headers=headers, params=params)
+        if not resp.status_code == 200:
```

Comment:
Just a note: usually *not equal* is written like `if resp.status_code != 200`, but it is not a crime to write like this.